### PR TITLE
Fixed PG::InvalidColumnReference with DISTINCT & by_distance

### PR DESCRIPTION
**Pull request by elsurudo**
I am using PG and Rails 5.1.

I had a pretty complex statement that used both `.distinct` and the `.by_distance` scope, and I was getting this error:

PG::InvalidColumnReference: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list

I dug down a bit and found that it was happening because the formula was right in the ORDER BY clause, and Postgres was having none of it.

This PR is a fix that worked for me. Whether it's something should be merged in, I don't know. But it did fix a bug for me, and it may for others, as well. If there is some further refactoring I could do to get this merged, let me know.

Basically, I am looking for feedback.

### CODE REVIEW

**Comment by elsurudo** [comment]
Looks like `self.all` is a problem because it returns an array instead of a relation. Confused because `scoped` was removed from Rails. I'm sure there is a way to fix this.

**Comment by mnoack** [comment]
I'm closing until we fix this. But I'm absolutely happy to merge Pull Requests which pass the tests, maintains (or improves test coverage) and the code itself seems reasonable.

**Comment by elsurudo** [comment]
It's been a long time since I created this so I forgot much of the context. Your main objection, in that case, is that it decreased the test coverage, right?

**Comment by onwardmk** [comment]
I am experiencing this issue. Is there a workaround?
